gnomad/utils/constraint.py

```diff
@@ -0,0 +1,162 @@
+# noqa: D100
+# cSpell: disable
+from typing import Any, Optional, Tuple, Union
+import logging
+
+import hail as hl
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("constraint_pipeline")
+logger.setLevel(logging.INFO)
+
+
+def annotate_with_mu(
+    ht: hl.Table,
+    mutation_ht: hl.Table,
+    output_col: str = "mu_snp",
+) -> hl.Table:
+    """
+    Annotate SNP mutation rate for the input Table.
+
+    :param ht: Input Table.
+    :param mutation_ht: Mutation rate Table.
+    :param output_col: Name for mutational rate annotation. Defaults to 'mu_snp'.
+    :return: Table with mutational rate annotation added (default name for annotation is 'mu_snp').
+    """
+    keys = mutation_ht.key
+    mu = hl.literal(
```

Comment:
The performance doesn't make much difference for the suggested code below. However, I didn't make any change because I can't `count` or `show` the table generated by your code. It stuck at the beginning, and I can't `anti_join` the old and new table. Does it happen to you?
```suggestion
    mu = mutation_ht.index(*[ht[k] for k in mutation_ht.key]).mu_snp
    return ht.annotate(
        **{output_loc: hl.case().when(hl.is_defined(mu), mu).or_error("Missing mu")}
    )
```